### PR TITLE
Fix NPE for DDAgentWriter default builder with default monitoring

### PR DESCRIPTION
**Pull request by drodriguezhdez**
Fix NPE for DDAgentWriter default builder with default monitoring

### CODE REVIEW

**Comment by richardstartin** [comment]
Does this affect 0.62.0?

File: dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java [hunk 1]
```diff
@@ -85,8 +86,8 @@ private DDAgentWriter(
               monitoring);
     }
     this.healthMetrics = healthMetrics;
-    this.dispatcher = new PayloadDispatcher(api, healthMetrics, monitoring);
```
Comment (richardstartin) [inline dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java:88]: I would really prefer we don't make changes like these

Comment (drodriguezhdez) [reply dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java:88]: You mean the `this.` prefix, right?

**Comment by drodriguezhdez** [comment]
Yes, the 0.62.0 is affected by this.

File: dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java (hunk 1 above)
Comment (richardstartin) [reply dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java:88]: yes, I imagine this happened as a result of some IDE setting?

Comment (drodriguezhdez) [reply dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java:88]: Yes, changing 👍

**Comment by richardstartin** [comment]
What actual impact does this have? `DDAgentWriter` is built in `CoreTracer` and is provided a `Monitoring` implementation there.

File: dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java (hunk 1 above)
Comment (richardstartin) [reply dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java:88]: yes, I imagine this happened as a result of some IDE setting?

**Comment by drodriguezhdez** [comment]
If the user creates the `DDTracer` manually, which is what we're doing in the Jenkins-plugin, this error appears.

File: dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java (hunk 1 above)
Comment (drodriguezhdez) [reply dd-trace-core/src/main/java/datadog/trace/common/writer/DDAgentWriter.java]: Changed

**Review by richardstartin — APPROVED** [review]
Fair enough. I think we need to review our public API though.

**Comment by richardstartin** [comment]
Could you do yourself a favour and add a test for the exact way you are constructing the tracer so that it would break next time? It would also help to offer a migration path if we end up removing the API you are using.

**Comment by drodriguezhdez** [comment]
Yes, I'm adding it right now 👍

**Review by mcculls — APPROVED** [review]

**Pull request merged**